text/0000-vec-fallible-allocation.md

```diff
@@ -0,0 +1,341 @@
+and the accepted [fallible_collection_alloc feature](https://github.com/rust-lang/rfcs/blob/master/text/2116-alloc-me-maybe.md).
+
+As a brief summary, there are environments where dynamic allocation is required and failing to allocate is both expected and able to be handled. For example,
+in OS Kernel environments (such as [Linux](https://lore.kernel.org/lkml/CAHk-=wh_sNLoz84AUUzuqXEsYH35u=8HV3vK-jbRbJ_B-JjGrg@mail.gmail.com/)), some embedded
+systems, high-reliability systems (such as databases) and multi-user services (where a single request may fail without the entire service halting, such as a
+web server).
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+One often unconsidered failure in a software system is running out of memory (commonly referred to as "Out-of-memory" or OOM):
+if code attempts to dynamically allocate memory, and there is an insufficient amount of available memory on the system, how should this be handled?
+
+In most applications the failure to allocate means that the code cannot make any progress, and so the appropriate response is to exit immediately indicating that there was a failure (i.e., to `panic`).
+Consider the Rust Compiler itself, if the compiler cannot allocate memory then it cannot fail that part of the compilation and attempt to continue the rest of the compilation.
+The only appropriate action is to fail the entire compilation and emit an error message.
+
+Since it is appropriate for most circumstances, the Rust standard library defaults to panicking on allocation failure.
+This approach is referred to as "infallible allocation": from the perspective of a function caller the allocation can never fail, because if it does then the called function will `panic` and the caller won't live to observe the failure.
+
+There are, however, environments where allocation failures are both expected and required to be handled more gracefully.
+Consider a multi-user service such as a database or a web server: if a request comes in to the service that requires allocating more memory than is available on the system, then the service can respond with a failure for just that request, and it can continue to service other requests.
+This approach is referred to as "fallible allocation": each function that may allocate needs to indicate to its caller if an attempt to allocate has failed, and that caller must explicitly handle that possibility.
+
+Within the Rust standard library, one can identify "fallible allocation" functions by their names being prefixed with `try_` and their return types being an instance of `Result`.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Currently any function in `alloc` that may all the global OOM handler (which panics) rather than let its caller handle all allocation failures is marked with `#[cfg(not(no_global_oom_handling))]` (see <https://github.com/rust-lang/rust/pull/84266>),
```

Comment:
```suggestion
Currently any function in `alloc` that may call the global OOM handler (which panics) rather than let its caller handle all allocation failures is marked with `#[cfg(not(no_global_oom_handling))]` (see <https://github.com/rust-lang/rust/pull/84266>),
```